### PR TITLE
Upgrade minimum version of woodwork to v0.11.0

### PR DESCRIPTION
**Pull request by angela97lin**

### CODE REVIEW

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -11,10 +11,12 @@ Release Notes
         * ``TimeSeriesParametersDataCheck`` was added to ``DefaultDataChecks`` for time series problems :pr:`3139`
         * Renamed ``date_index`` to ``time_index`` in ``problem_configuration`` for time series problems :pr:`3137`
         * Updated ``nlp-primitives`` minimum version to 2.1.0 :pr:`3166`
+        * Updated minimum version of ``woodwork`` to v0.11.0 :pr:`3171`
     * Documentation Changes
         * Added comments to provide clarity on doctests :pr:`3155`
     * Testing Changes
         * Parameterized tests in ``test_datasets.py`` :pr:`3145`
+
```
Comment (jeremyliweishih) [inline docs/source/release_notes.rst:19]: extra line?

File: evalml/tests/pipeline_tests/test_pipeline_utils.py [hunk 2]
```diff
@@ -143,32 +143,19 @@ def test_make_pipeline(
                 else []
             )
             natural_language_featurizer = (
-                [NaturalLanguageFeaturizer]
-                if "text" in column_names and input_type == "ww"
-                else []
+                [NaturalLanguageFeaturizer] if "text" in column_names else []
```
Comment (jeremyliweishih) [inline evalml/tests/pipeline_tests/test_pipeline_utils.py:146]: why is this needed?

**Review by jeremyliweishih — APPROVED** [review]
LGTM once tests pass!

File: evalml/tests/pipeline_tests/test_pipeline_utils.py (hunk 2 above)
Comment (angela97lin) [reply evalml/tests/pipeline_tests/test_pipeline_utils.py:146]: New natural language inference means even with pandas input, we'll infer natural language! Before, it was inferred as Unknown, hence the DropColumn transformer that got removed.

Comment (jeremyliweishih) [reply evalml/tests/pipeline_tests/test_pipeline_utils.py:146]: sounds great, thanks!

**Review by bchen1116 — APPROVED** [review]
LGTM!
